csrc/ops/alias.cpp

```diff
@@ -1124,4 +1124,88 @@ TensorView* expand_as(TensorView* inp, TensorView* other) {
   return out_tensor;
 }
 
+TensorView* repeat(TensorView* inp_tv, std::vector<int64_t> repeat_times) {
+  const auto ndims =
+      TensorDomain::noReductions(inp_tv->getLogicalDomain()).size();
+
+  // Handle repetitions of non-broadcast IDs first. Each ID is
+  // individully repeated by:
+  //
+  // Step 1. Insert a broadcast ID immediately outside of the
+  // repeated ID
+  // Step 2. Expand the broadcast ID by the repetition factor
+  // Step 3. Flatten the expanded ID and the repeated ID
+
+  bool has_repetition_of_broadcast = false;
+  auto intermediate_tv = inp_tv;
+  for (const auto i : c10::irange(ndims)) {
```

Comment:
As a potential improvement for fusion IR readability, we don't have to handle one dimension at a time, do we? For example, `repeat([M, N], [2, 3])` can be done as 
```
broadcast([M, N], [True, False, True, False])
expand([1, M, 1, N], [2, 1, 3, 1])
view([2, M, 3, N], [2M, 3N])
```

Reply:
You're right. That's possible. Added a comment.